### ISSUE TITLE
CMake version detection not robust enough

### ISSUE THREAD

**Issue by rezahousseini** [mesonbuild/meson#7093]
If I use cmake binary from the [mxe](https://mxe.cc) project the following output gets printed when calling `x86_64-w64-mingw32-static-cmake --version`

```
== Using MXE wrapper: /usr/bin/x86_64-w64-mingw32.static-cmake
     - cmake version 3.15.4
     - warnings for unused CMAKE_POLICY_DEFAULT variables can be ignored
== Using MXE toolchain: /usr/x86_64-w64-mingw32.static/share/cmake/mxe-conf.cmake
== Using MXE runresult: /usr/share/cmake/modules/TryRunResults.cmake
== Adding "-DCMAKE_BUILD_TYPE=Release"
loading initial cache file /usr/share/cmake/modules/TryRunResults.cmake
CMake Error at /usr/share/cmake/modules/TryRunResults.cmake:1 (SET):
  Unknown CMake command "SET".


cmake version 3.15.4

CMake suite maintained and supported by Kitware (kitware.com/cmake).
```

This clearly can't be detected by

```
p, out = Popen_safe(cmakebin.get_command() + ['--version'])[0:2]
cmvers = re.sub(r'\s*(cmake|cmake3) version\s*', '', out.split('\n')[0]).strip()
```

which is the code in the `check_cmake` function on line 115 in the [executor.py](https://github.com/mesonbuild/meson/blob/bfea80677eef52271cde1c089268a6d544a79453/mesonbuild/cmake/executor.py) file

**Comment by rezahousseini** [comment]
A working regexp would be

    re.search(r'(cmake|cmake3)\s*version\s*([\d.]+)', out).group(2)

**Comment by rezahousseini** [comment]
I created a pull request https://github.com/mesonbuild/meson/pull/7095

**Comment by mensinda** [comment]
Should be fixed, since #7095 is in. Closing.
